### DISCUSSION TITLE
Pocketbase server status

### DISCUSSION THREAD

**Discussion by sabikrahat** [pocketbase/pocketbase#1957] [Q&A]
Is there any way to check pocketbase server is running or not? I want to show a connected button which will automatically become red when the server turns off. Is there any way to get server status as stream in dart?

**Comment by ganigeorgiev** [comment]
There is no dedicated realtime health check if that's what you are looking for.

There is a [`GET /api/health` endpoint](https://pocketbase.io/docs/api-health/) (`await pb.health.check()`) that will return 200 if the server is up but you have to ping it manually.

**Reply by ganigeorgiev** [reply]
Alternatively, you can also create a "blank" realtime connection using the package SSE client and use its stream or `onClose` callback to detect whether it is connected or disconnected. You can find an example how to initialize it in the dartdoc of the client: https://github.com/pocketbase/dart-sdk/blob/master/lib/src/sse/sse_client.dart.

**Reply by sabikrahat** [reply]
Can you please give an example of  SSE Client in dart?

**Reply by ganigeorgiev** [reply]
@sabikrahat Actually I now remember that the `SseClient` is not exported and you can't access it directly (_and I'm not sure if it will be a good idea to export it since it'll increase unnecessary the API surface_). You have to manually ping the health endpoint.

**Reply by ganigeorgiev** [reply]
If you really want to import it, you can try something like below (but future backward compatibility is not guaranteed!):

```dart
import "package:pocketbase/src/sse/sse_client.dart";

final sse = SseClient(
  "https://example.com/api/realtime",
  onClose: () => {print("on final disconnect...")},
  onError: (err) => {print("on reconnect attempt (will happen on server idle reconnect or no internet connectivity)")},
);

// you can eventually attach to the onMessage stream (but for this particular use case you may not need it since you don't have any subscriptions)
sse.onMessage.listen((msg) {
  print(msg);
});

// close the connection and clean up any resources associated with it
sse.close();
```

**Reply by nicetomytyuk** [reply]
@ganigeorgiev Hi, Is there a way to know when the realtime connection is reestablished? I need to fetch new data when realtime connection is back...

**Reply by ganigeorgiev** [reply]
@nicetomytyuk If you use the Dart SDK, you can try listening to the `PB_CONNECT` event:

```dart
pb.realtime.subscribe("PB_CONNECT", (e) {
  print("connected: $e");
});
```

Note that the JS SDK currently doesn't have this available (it will be added with the next release).

**Reply by nicetomytyuk** [reply]
@ganigeorgiev Oh I will wait for the next release, I'm using JS SDK at the moment, Is there a workaround I can use for now?

**Reply by ganigeorgiev** [reply]
@nicetomytyuk I've pushed a minor fix to propagate the `PB_CONNECT` event to the subscriptions with [JS SDK v0.20.1](https://github.com/pocketbase/js-sdk/releases/tag/v0.20.1) (_it will be available shortly in the npm registry_).

**Reply by nicetomytyuk** [reply]
@ganigeorgiev just gave it a try, that's what I was looking for, thank you!
